### PR TITLE
Consistent bit-width for activations and weight in propagation mode

### LINKED ISSUE #144: Mixed Precision missquantization
While training object detection model through hawq config, I realized there are much more int8 activation quantizations then int8 weight quantizations. According to [netron](https://lutzroeder.github.io/netron/) some convolutions take int8 activations and int4 weights. I suppose that's not how things should be. What do you think?
image

### PR DESCRIPTION
**Pull request by ljaljushkin**
Fixes #144 

Example of a consistent grouping of weight and activation quantizers for Inception-v3 (quantizers circled yellow are from the same group and should have the same bit-width): 

inception

@SKholkin, feel free to review bit-width graphs for other models from `tests\data\reference_graphs\quantized\hawq` folder
@RikAllen, next step is to introduce `allow_different_bitwidth_for_weight_and_activation` (#164) option to HW Config for independent handling bit-width inside group in HAWQ algorithm. 
@vuiseng9, this grouping is implemented for propagation-based mode only:
```json
'quantizer_setup_type': 'propagation_based'
```

### CODE REVIEW

**Comment by ljaljushkin** [comment]
Jenkins please retry a build

**Review by asenina — APPROVED** [review]

File: nncf/nncf_network.py [hunk 1]
```diff
@@ -74,12 +74,14 @@ class InsertionInfo:
     def __init__(self, op_exec_context: OperationExecutionContext,
                  is_input=False,
                  is_output=False,
-                 shape_to_operate_on=None):
+                 shape_to_operate_on=None,
+                 adjacent_quantizers_locations: 'AdjacentQuantizersLocations' = None):
```
Comment (vshampor) [inline nncf/nncf_network.py:78]: I am against this kind of extension. The new field is quantizer algo-specific, while the rest of the insertion info is more or less compression algorithm-agnostic. Either subclass the `InsertionInfo` struct, or use another means of passing the info between the propagation stage and the quantizer setup stage.

The naming should also be improved. It is not immediately obvious what kind of "adjacency" is meant in the name.

**Review by vshampor — CHANGES_REQUESTED** [review]

File: nncf/nncf_network.py (hunk 1 above)
Comment (ljaljushkin) [reply nncf/nncf_network.py]: Subclassed and renamed

File: nncf/quantization/quantizer_propagation.py [hunk 2]
```diff
@@ -214,8 +232,10 @@ def __init__(self, ip_graph: InsertionPointGraph, ignored_scopes=None):
                     self.QUANTIZATION_TRAIT_NODE_ATTR] = QuantizationTrait.NON_QUANTIZABLE
                 qpg_node[self.AFFECTING_PROPAGATING_QUANTIZERS_ATTR] = []
                 qpg_node[self.OPERATOR_METATYPE_NODE_ATTR] = node[InsertionPointGraph.OPERATOR_METATYPE_NODE_ATTR]
-                scope_node = str(node[InsertionPointGraph.REGULAR_NODE_REF_NODE_ATTR][
-                    NNCFGraph.OP_EXEC_CONTEXT_NODE_ATTR].input_agnostic)
+                iap_context = node[InsertionPointGraph.REGULAR_NODE_REF_NODE_ATTR][
+                    NNCFGraph.OP_EXEC_CONTEXT_NODE_ATTR].input_agnostic
+                qpg_node[self.OPERATOR_SCOPE] = iap_context.scope_in_model
+                scope_node = str(iap_context)
```
Comment (vshampor) [inline nncf/quantization/quantizer_propagation.py:238]: `node_scope`

**Review by vshampor — CHANGES_REQUESTED** [review]

File: nncf/quantization/quantizer_propagation.py (hunk 2 above)
Comment (ljaljushkin) [reply nncf/quantization/quantizer_propagation.py:238]: any other comments to make it worth re-running pre-commit validation?
I can fix it in my next PR's, it looks like a very minor change

Comment (vshampor) [reply nncf/quantization/quantizer_propagation.py:238]: Not yet, but there will be once I compare this PR and #168; let's wait until then

Comment (ljaljushkin) [reply nncf/quantization/quantizer_propagation.py:238]: @vshampor any update since then?

**Pull request merged**